arangod/Pregel/Actor/include/Actor/ActorList.h

```diff
@@ -0,0 +1,133 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Markus Pfeiffer
+/// @author Julia Volmer
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "Actor/ActorBase.h"
+#include "Actor/ActorPID.h"
+#include "Basics/Guarded.h"
+
+namespace arangodb::pregel::actor {
+
+struct ActorList {
+  struct ActorMap : std::unordered_map<ActorID, std::shared_ptr<ActorBase>> {};
```

Comment:
Do you really want to derive? Or would this suffice:
```suggestion
  using ActorMap = std::unordered_map<ActorID, std::shared_ptr<ActorBase>>;
```

Reply:
I used inheritance because I need a specific inspector for ActorMap because the ActorBase inside the shared pointer is an abstract class and therefore have to define the inspection explicitly here.